File: P115-PersistData/NSKeyedArchiverOCDemo/NSKeyedArchiverOCDemo/Person.m
//
//  Person.m
//  NSKeyedArchiverOCDemo
//
//  Created by Andy Ron on 2018/8/23.
//  Copyright © 2018年 Andy Ron. All rights reserved.
//

#import "Person.h"

@implementation Person

//解档
- (id)initWithCoder:(NSCoder *)aDecoder {
    if ([super init]) {
        self.name = [aDecoder decodeObjectForKey:@"name"];
        self.sex = [aDecoder decodeObjectForKey:@"sex"];
        self.age = [aDecoder decodeIntegerForKey:@"age"];
    }
    
    return self;
}

//归档
- (void)encodeWithCoder:(NSCoder *)aCoder {
    [aCoder encodeObject:self.name forKey:@"name"];
    [aCoder encodeObject:self.sex forKey:@"sex"];
    [aCoder encodeInteger:self.age forKey:@"age"];
}




@end

File: P115-PersistData/NSKeyedArchiverDemo/NSKeyedArchiverDemo/Person.swift
//
//  Person.swift
//  NSKeyedArchiverDemo
//
//  Created by Andy Ron on 2018/8/23.
//  Copyright © 2018年 Andy Ron. All rights reserved.
//

import Foundation

class Person: NSObject, NSCoding {
    var name: String
    var sex: String
    var age: Int
    
    init(name: String, sex: String, age: Int) {
        self.name = name
        self.sex = sex
        self.age = age
    }
    
    func encode(with aCoder: NSCoder) {
        aCoder.encode(self.name, forKey: "name")
        aCoder.encode(self.sex, forKey: "sex")
        aCoder.encode(self.age, forKey: "age")

    }
    
    required init?(coder aDecoder: NSCoder) {
        self.name = aDecoder.decodeObject(forKey: "name") as! String
        self.sex = aDecoder.decodeObject(forKey: "sex") as! String
        self.age = aDecoder.decodeObject(forKey: "age") as! Int
        
    }
    
    
}
